src/lib.cpp

```diff
@@ -0,0 +1,24 @@
+#include "lib.h"
+
+int myfun(bool b, int (*callback)(int)) {
+  int num = -1;
+
+  if (b) {
+    num = callback(num);
+  } else {
+    return callback(callback(num));
+  }
```

Comment:
```suggestion
  if (b) {
    num = callback(callback(num));
  } else {
    return num;
  }
```
To trigger the nullptr dereference on line 18, need to remove a call to `callback` at least from one branch.